### PR TITLE
Use `auto` for iterator variable types

### PR DESCRIPTION
**Pull request by gaul**
Applied via `clang-tidy -fix`.

### CODE REVIEW

File: src/addhead.cpp [hunk 1]
```diff
@@ -164,7 +164,7 @@ bool AdditionalHeader::AddHeader(headers_t& meta, const char* path) const
     // [NOTE]
     // Because to allow duplicate key, and then scanning the entire table.
     //
-    for(addheadlist_t::const_iterator iter = addheadlist.begin(); iter != addheadlist.end(); ++iter){
+    for(auto iter = addheadlist.begin(); iter != addheadlist.end(); ++iter){
```
[the remark this answers is not in the document]
Comment (gaul) [reply src/addhead.cpp:167]: Done.

**Comment by gaul** [comment]
There is an interesting issue with `auto it = map.find()` that loses `const` without an explicit `const_iterator` type.  I didn't find any regressions in s3fs though.

**Comment by gaul** [comment]
CentOS 7 seems to lack the ability to do `std::vector::erase(const_iterator)`.

File: src/addhead.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (ggtakec) [reply src/addhead.cpp]: Thanks

**Comment by ggtakec** [comment]
@gaul 
> CentOS 7 seems to lack std::vector::erase(const_iterator).

Does that mean I'll get compilation errors on CentOS7?
(Should we wait to merge this PR?)

**Comment by gaul** [comment]
This PR is ready to merge.  I had to remove some calls to `cbegin` and `cend` due to CentOS 7 limitations.

**Review by ggtakec — APPROVED** [review]

**Pull request merged**